kornia/utils/pointcloud_io.py

```diff
@@ -81,3 +81,64 @@ def load_pointcloud_ply(filename: str, header_size: int = 8) -> torch.Tensor:
         # create tensor from list
         pointcloud: torch.Tensor = torch.tensor(points)
         return pointcloud
+
+
+def iterative_closest_point(
+    points_in_a: torch.Tensor, points_in_b: torch.Tensor, max_iterations: int = 20, tolerance: float = 1e-4
+) -> torch.Tensor:
+    """Compute the relative transformation between two point clouds.
+
+    The resulting transformation uses the iterative closest point algorithm to satisfy:
+
+        points_in_b = b_from_a @ points_in_a
+
+    Args:
+        points_in_a: The point cloud in the source coordinates frame A  with shape Nx3
+        points_in_b:  The point cloud in the source coordinates frame A with shape Nx
+        max_iterations (int): Maximum number of iterations to run.
+        tolerance (float): Tolerance criteria for stopping.
+
+     Return:
+        The relative transformation between the two pointcloud with shape 3x4
+    """
+
+    src = points_in_a.clone()
```

Comment:
add some basic shape and type check 
`KORNIA_CHECK_SHAPE(points_in_a, ["B", "3"])`